stl/inc/random

```diff
@@ -2472,9 +2473,10 @@ private:
             for (;;) { // generate and reject
                 _Ty1 _Yx;
                 for (;;) { // generate a tentative value
-                    _Yx  = static_cast<_Ty1>(_CSTD tan(_Pi * _NRAND(_Eng, _Ty1)));
-                    _Res = static_cast<_Ty>(_Par0._Sqrt * _Yx + _Par0._Mean);
-                    if (_Ty{0} <= _Res && _Res <= _Par0._Tx) {
+                    _Yx = static_cast<_Ty1>(_CSTD tan(_Pi * _NRAND(_Eng, _Ty1)));
+                    const _Ty1 _Mx{_Par0._Sqrt * _Yx + _Par0._Mean};
+                    if (0.0 <= _Mx && _Mx < _Par0._Tx + _Ty{1}) {

```

Comment:
`_Par0._Tx + _Ty{1}` could be UB (signed `_Ty`) or wrap to zero (unsigned `_Ty`) when `_Par0._Tx` is `numeric_limits<_Ty>::max()`.

Reply:
Thanks, this should be fixed as part of #1123